src/utils/dex/evolution.js

```diff
@@ -235,6 +236,187 @@ function validIds(path, ar) {
   });
 }
 
+function genForms() {
+  const formNamedata = formPokemonNames;
+  const formsList = formNamedata["labelDataArray"];
+  const forms = {};
+
+  for (let i = 0; i < formsList.length; i++) {
+    const allForms = formsList[i];
+    if (allForms["arrayIndex"] !== 0 && parseInt(allForms["labelName"].slice(-3)) > 0) {
```

Comment:
There's a lot going on in this expression, can you extract it to a named variable?